packages/core/src/modules/vc/W3cCredentialService.ts

```diff
@@ -186,6 +187,35 @@ export class W3cCredentialService {
     return w3cCredentialRecord
   }
 
+  /**
+   * Revoke a credential by issuer
+   * associated with the credential record.
+   *
+   * @param credentialRecordId The id of the credential record for which to revoke the credential
+   * @returns Revoke credential notification message
+   *
+   */
+  public async revokeCredential<Format extends ClaimFormat.JwtVp | ClaimFormat.LdpVp>(
+    agentContext: AgentContext,
+    options: RevokeCredentialOptions
+  ) {
+    const credentialRecod = await this.getCredentialRecordById(agentContext, options.credentialRecordId)
```

Comment:
- I think we should not use the credentialRecord to get tags as credentialRecord may or may not be present in wallet. To get the claimFormat we can use options

Reply:
Ahh yes you are right. Also as the specs suggest the issuer may not always be the one revoking the credential. Will change it